packages/oui-radio/src/index.spec.js

```diff
@@ -1,24 +1,179 @@
+    beforeEach(inject((_TestUtils_) => {
+        TestUtils = _TestUtils_;
     }));
 
-    describe("Controller", () => {
-        it("should exist", () => {
-            const ctrl = $componentController("ouiRadio", {
-                $element: {},
-                $attrs: {},
-                $transclude: {}
+    const getRadioInputElement = (element) => element[0].querySelector("input[type=radio]");
+    const getRadioLabelElement = (element) => element[0].querySelector("label");
+    const getRadioTextContainerElement = (element) => element[0].querySelector(".oui-radio__label span:first-child");
+    const getRadioDescriptionElement = (element) => element[0].querySelector(".oui-radio__description");
+
+    describe("Component", () => {
+        describe("id attribute", () => {
+            it("should generate an id for the input and label when undefined", () => {
+                const element = TestUtils.compileTemplate("<oui-radio></oui-radio>");
+
+                const radioElement = getRadioInputElement(element);
+                expect(angular.element(radioElement).prop("id")).toMatch(/^oui-radio-\d+$/);
+
+                const radioLabelElement = getRadioLabelElement(element);
+                expect(angular.element(radioLabelElement).attr("for")).toMatch(/^oui-radio-\d+$/);
+            });
+
+            it("should set the id for the input and label when defined", () => {
+                const element = TestUtils.compileTemplate('<oui-radio id="test"></oui-radio>');
+
+                const radioElement = getRadioInputElement(element);
+                expect(angular.element(radioElement).prop("id")).toBe("test");
+
+                const radioLabelElement = getRadioLabelElement(element);
+                expect(angular.element(radioLabelElement).attr("for")).toBe("test");
             });
-            expect(ctrl).toBeDefined();
         });
-    });
 
-    describe("Component", () => {
-        xit("should display a radio");
+        describe("name attribute", () => {
+            it("should set the name attribute on input when defined", () => {
+                const element = TestUtils.compileTemplate('<oui-radio name="test"></oui-radio>');
+
+                const radioElement = getRadioInputElement(element);
+                expect(angular.element(radioElement).prop("name")).toBe("test");
+            });
+        });
+
+        describe("text attribute", () => {
+            it("should display a text inside the radio's text container", () => {
+                const element = TestUtils.compileTemplate('<oui-radio text="test"></oui-radio>');
+
+                const textContainerRadioElement = getRadioTextContainerElement(element);
+                expect(angular.element(textContainerRadioElement).html()).toBe("test");
+            });
+        });
+
+        describe("description attribute", () => {
+            it("should display the radio's description container when empty", () => {
```

Comment:
I think it should be "should not display a text inside the radio's description container" to match the assertion